standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreUtils.java

```diff
@@ -38,4 +38,19 @@ public static void logAndThrowMetaException(Exception e) throws MetaException {
     throw new MetaException(exInfo);
   }
 
+  public static String encodeTableName(String name) {
```

Comment:
I think this method might be moved to Warehouse..."in case we want to keep the automatic location filling feature".

An alternative can be to force all metastore clients to fill out the location during every meta object creation ( and put a guard on it ) - if we do that; we can leave this "encoder" behind.